eliminate panic raising in parser

Compiling such sql cause to panic 
`TABLE MyTable1 INHERITS ODocUnknown (
    MyField ref(registry.Login) NOT NULL
);
`

Error stack trace:

panic: unsupported def kind 0

goroutine 23 [running]:
github.com/voedger/voedger/pkg/parser.(*buildContext).isExists(0x140006571a8?, {{0x100d833f0?, 0x14000657248?}, {0x140002480d0?, 0x140006571b8?}}, 0x20?)
	/Users/alisher/projects/untill/voedger/pkg/parser/impl_build.go:907 +0x1bc
github.com/voedger/voedger/pkg/parser.(*buildContext).table(0x14000598ff0, 0x100a5b5c4?, 0x140001b8630, 0x1400014fce0?)
	/Users/alisher/projects/untill/voedger/pkg/parser/impl_build.go:611 +0x58
github.com/voedger/voedger/pkg/parser.(*buildContext).tables.func1(0x14000657288?, 0x100a5beac?)
	/Users/alisher/projects/untill/voedger/pkg/parser/impl_build.go:574 +0x38
github.com/voedger/voedger/pkg/parser.iteratePackageStmt[...].func1(0x140006572b8?)
	/Users/alisher/projects/untill/voedger/pkg/parser/utils.go:177 +0x44
github.com/voedger/voedger/pkg/parser.iterateContext.func1({0x100e9b1c0, 0x140001b8630})
	/Users/alisher/projects/untill/voedger/pkg/parser/utils.go:184 +0x3c
github.com/voedger/voedger/pkg/parser.(*SchemaAST).Iterate(0x1400009e990, 0x1400016a048)
	/Users/alisher/projects/untill/voedger/pkg/parser/types.go:82 +0x40
github.com/voedger/voedger/pkg/parser.iterateContext(0x14000489e60, 0x1400016a030)
	/Users/alisher/projects/untill/voedger/pkg/parser/utils.go:183 +0x84
github.com/voedger/voedger/pkg/parser.iteratePackage(...)
	/Users/alisher/projects/untill/voedger/pkg/parser/utils.go:170
github.com/voedger/voedger/pkg/parser.iteratePackageStmt[...](0x14000489350, 0x14000598ff0, 0x1400016a018)
	/Users/alisher/projects/untill/voedger/pkg/parser/utils.go:175 +0xe8
github.com/voedger/voedger/pkg/parser.(*buildContext).tables(0x14000598ff0)
	/Users/alisher/projects/untill/voedger/pkg/parser/impl_build.go:573 +0x130
github.com/voedger/voedger/pkg/parser.(*buildContext).build(0x14000598ff0)
	/Users/alisher/projects/untill/voedger/pkg/parser/impl_build.go:50 +0x164
github.com/voedger/voedger/pkg/parser.buildAppDefs(0x1400014f398, {0x100ee89e8?, 0x140004eb0b0})
	/Users/alisher/projects/untill/voedger/pkg/parser/impl.go:316 +0xbc
github.com/voedger/voedger/pkg/parser.BuildAppDefs(...)
	/Users/alisher/projects/untill/voedger/pkg/parser/provide.go:49
github.com/voedger/voedger/cmd/vpm.compile({0x100ee0da8, 0x14000220600}, {0x140001eee04, 0x69})
	/Users/alisher/projects/untill/voedger/cmd/vpm/compile.go:84 +0x8ec
github.com/voedger/voedger/cmd/vpm.newCompileCmd.func1(0x14000216100?, {0x140001909c0?, 0x4?, 0x100d73275?})
	/Users/alisher/projects/untill/voedger/cmd/vpm/compile.go:42 +0xe8
github.com/spf13/cobra.(*Command).execute(0x14000244000, {0x14000190980, 0x1, 0x1})
	/Users/alisher/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:940 +0x658
github.com/spf13/cobra.(*Command).ExecuteC(0x14000244300)
	/Users/alisher/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:1068 +0x320
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/alisher/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:992
github.com/spf13/cobra.(*Command).ExecuteContext(...)
	/Users/alisher/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:985
github.com/untillpro/goutils/cobrau.ExecCommandAndCatchInterrupt.func1({0x100ee3ad0?, 0x140001d45a0?})
	/Users/alisher/go/pkg/mod/github.com/untillpro/goutils@v0.0.0-20231117130619-11fe1908c171/cobrau/catch_interrupt.go:21 +0x58
github.com/untillpro/goutils/cobrau.goAndCatchInterrupt.func1()
	/Users/alisher/go/pkg/mod/github.com/untillpro/goutils@v0.0.0-20231117130619-11fe1908c171/cobrau/catch_interrupt.go:40 +0x70
created by github.com/untillpro/goutils/cobrau.goAndCatchInterrupt in goroutine 20
	/Users/alisher/go/pkg/mod/github.com/untillpro/goutils@v0.0.0-20231117130619-11fe1908c171/cobrau/catch_interrupt.go:38 +0x148


Process finished with the exit code 1

--- a/pkg/parser/impl_analyse.go
+++ b/pkg/parser/impl_analyse.go
@@ -667,15 +667,17 @@ func getTableInheritanceChain(table *TableStmt, c *iterateCtx) (chain []tableNod
 	vf = func(t *TableStmt) error {
 		if t.Inherits != nil {
 			inherited := *t.Inherits
-			if err := resolveInCtx(inherited, c, func(t *TableStmt, pkg *PackageSchemaAST) error {
+			t, pkg, err := lookupInCtx[*TableStmt](inherited, c)
+			if err != nil {
+				return err
+			}
+			if t != nil {
 				node := tableNode{pkg: pkg, table: t}
 				if refCycle(node) {
 					return ErrCircularReferenceInInherits
 				}
 				chain = append(chain, node)
 				return vf(t)
-			}); err != nil {
-				return err
 			}
 		}
 		return nil

--- a/pkg/parser/impl_test.go
+++ b/pkg/parser/impl_test.go
@@ -1919,3 +1919,18 @@ TABLE SomeTable INHERITS CDoc (
 	}, "\n"))
 
 }
+
+func Test_ODocUnknown(t *testing.T) {
+	require := require.New(t)
+	pkgApp1 := buildPackage(`APPLICATION registry();
+TABLE MyTable1 INHERITS ODocUnknown ( MyField ref(registry.Login) NOT NULL ); 
+`)
+
+	_, err := BuildAppSchema([]*PackageSchemaAST{pkgApp1, getSysPackageAST()})
+	require.EqualError(err, strings.Join([]string{
+		"source.sql:2:1: undefined table kind",
+		"source.sql:2:39: registry undefined",
+		"source.sql:2:1: ODocUnknown undefined",
+	}, "\n"))
+
+}

